packages/core/src/render3/node_manipulation.ts

```diff
@@ -166,11 +167,16 @@ function walkTNodeTree(
  */
 function executeNodeAction(
     action: WalkTNodeTreeAction, renderer: Renderer3, parent: RElement | null,
-    node: RComment | RElement | RText, beforeNode?: RNode | null) {
+    node: RComment | RElement | RText, tNode: TNode, beforeNode?: RNode | null) {
   if (action === WalkTNodeTreeAction.Insert) {
     nativeInsertBefore(renderer, parent !, node, beforeNode || null);
   } else if (action === WalkTNodeTreeAction.Detach) {
-    nativeRemoveChild(renderer, parent !, node);
+    // by determining whether or not we are removing a host element node
+    // it will help the renderer to decide whether anything special before
+    // removal (like calling the leave animation (if defined) on the element
+    // before it is removed from the page).
+    const isHostElement = (tNode.flags & TNodeFlags.isComponent) > 0;
```

Comment:
You can use the `isComponent` util here:

```suggestion
    nativeRemoveChild(renderer, parent !, node, isComponent(tNode));
```